cukinia

```diff
@@ -877,6 +883,66 @@ _cukinia_knoerror()
 	return $ret
 }
 
+# _cukinia_network_interface_up: check if a network interface is up
+# arg1: interface name
+_cukinia_network_interface_up() {
+	local interface="$1"
+
+	_cukinia_prepare "Checking if interface \"$interface\" is ${__not:+NOT }up"
+
+	ip link show "$interface" up >/dev/null 2>&1 && return 0 || return 1
+}
+
+# _cukinia_network_interface_ipv4: check if an interface has an IPv4 address
+# arg1: interface name
+_cukinia_network_interface_ipv4() {
+	local interface="$1"
+
+	_cukinia_prepare "Checking if interface \"$interface\" has ${__not:+NO }IPv4 address"
+
+	ip -4 -f inet addr show dev "$interface" | grep -q inet && return 0 || return 1
```

Comment:
From my man pages: "-4" is a shortcut for "-f inet"

Reply:
True, change the command options.